test/realm_test.dart

```diff
@@ -1394,6 +1396,173 @@ Future<void> main([List<String>? args]) async {
     expect(await realmIsCancelled, isTrue);
     expect(progressReturned, isFalse);
   });
+
+  const compactTest = "compact_test";
+  void addDataForCompact(Realm realm) {
+    realm.write(() {
+      for (var i = 0; i < 2500; i++) {
+        realm.add(Product(ObjectId(), compactTest));
+      }
+    });
+
+    realm.write(() => realm.deleteMany(realm.query<Product>("stringQueryField CONTAINS '$compactTest'")));
+
+    realm.write(() {
+      for (var i = 0; i < 10; i++) {
+        realm.add(Product(ObjectId(), compactTest));
+      }
+    });
+  }
+
+  Future<int> createRealmForCompact(Configuration config) async {
+    var realm = getRealm(config);
+    
+    if (config is FlexibleSyncConfiguration) {
+      realm.subscriptions.update((mutableSubscriptions) {
+        mutableSubscriptions.add(realm.query<Product>("stringQueryField CONTAINS '$compactTest'"));
+      });
+      await realm.subscriptions.waitForSynchronization();
+    }
+
+    addDataForCompact(realm);
+    
+    if (config is FlexibleSyncConfiguration) {
+      await realm.syncSession.waitForDownload();
+      await realm.syncSession.waitForUpload();
+    }
+
+    final beforeSize = await File(config.path).stat().then((value) => value.size);
+
+    realm.close();
+    return beforeSize;
+  }
+
+  void validateCompact(bool compacted, String realmPath, int before) async {
+    expect(compacted, true);
+    final afterSize = await File(realmPath).stat().then((value) => value.size);
+    expect(before, greaterThan(afterSize));
+  }
+
+  test('Realm - local realm can be compacted', () async {
+    var config = Configuration.local([Product.schema], path: p.join(Configuration.defaultStoragePath, "${generateRandomString(8)}.realm"));
+    final beforeCompact = await createRealmForCompact(config);
```

Comment:
Let's say that this is the size:  beforeCompactSize

Reply:
alright. as you wish.